validators/http/http.go

```diff
@@ -147,11 +202,24 @@ func (v *Validator) Validate(input interface{}, unused []byte) (bool, error) {
 			return false, nil
 		}
 	}
+
+	if failureHeader := v.c.GetFailureHeader(); failureHeader != nil {
+		if lookupHTTPHeader(res.Header, failureHeader.GetName(), v.failureHeaderRegexp) {
+			return false, nil
+		}
+	}
+
 	if v.c.GetSuccessStatusCodes() != "" {
 		if lookupStatusCode(res.StatusCode, v.successStatusCodeRanges) {
 			return true, nil
 		}
```

Comment:
Instead of returning true here and below, I think we should return early only if a condition is not met, i.e. we should true only if all conditions are met. I'd do something like:

```
if v.c.GetSuccessStatusCodes() != "" {
		if !lookupStatusCode(res.StatusCode, v.successStatusCodeRanges) {
			return false, nil
		}
}
...
...

return true, nil
```

Reply:
Changed